tests/trickle/main.c

```diff
@@ -45,13 +61,28 @@ static trickle_t trickle = { .callback.func = &callback,
 int main(void)
 {
     msg_t msg;
+    bool reset = false;
+    unsigned counter = 0;
 
     msg_init_queue(_msg_q, Q_LEN);
 
     trickle_start(sched_active_pid, &trickle, TRICKLE_MSG, TR_IMIN,
                   TR_IDOUBLINGS, TR_REDCONST);
 
-    while (1) {
+    while (!error) {
+        if (!reset && (counter == FIRST_ROUND)) {
```

Comment:
I think you can omit the `reset` variable by  `#define SECOND_ROUND    (20U)` and change the `if else` to

```
if (counter ==  FIRST_ROUND) {
    prev_diff = 0;
    trickle_reset_timer(&trickle);
    puts("TRICKLE RESET");
}
else if (counter == SECOND_ROUND) {
    puts("[SUCCESS]");
    return 0;
}
```

Reply:
this only works if `FIRST_ROUND < SECOND_ROUND`. But I can adapt to this and  leave a note in the code

Reply:
yeah sure, it changes the logic how the vars are actually interpreted.